docs/en/02_Developer_Guides/03_Forms/01_Validation.md

```diff
@@ -163,9 +163,9 @@ class Page_Controller extends ContentController
         $check = Member::get()->filter('Email', $data['Email'])->first();
 
         if ($check) {
-            $form->addErrorMessage('Email', 'This email already exists', 'bad');
-
-            return $this->redirectBack();
+            $validationResult = new ValidationResult();
+            $validationResult->addFieldError('Email', 'This email already exists');
+            throw new ValidationException($validationResult);
```

Comment:
```suggestion
            $form->setSessionValidationResult($result);
            $form->setSessionData($form->getData());
            return $this->redirectBack();
```

Reply:
This exception will be handled by the `FormRequestHandler::httpSubmission()`
( https://github.com/silverstripe/silverstripe-framework/blob/bcab982aa7da2abd0061464dd5f96cf591a3c99a/src/Forms/FormRequestHandler.php#L251 ).
I have modified the code as you suggested. It looks better and is more readable.